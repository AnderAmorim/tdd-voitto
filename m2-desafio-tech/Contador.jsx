import React, { useState } from "react";

const Contador = () => {
  const [valor, setValor] = useState(0);

  const incrementar = () => {
    setValor(valor + 1);
  };

  const decrementar = () => {
    if (valor > 0) {
      setValor(valor - 1);
    }
  };

  return (
    <div>
      <h2>Contador</h2>
      <p>Valor: {valor}</p>
      <button onClick={incrementar}>Incrementar</button>
      <button onClick={decrementar}>Decrementar</button>
    </div>
  );
};

export default Contador;