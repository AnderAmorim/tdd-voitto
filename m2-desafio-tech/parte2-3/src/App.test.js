import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Contador from './App';

test('renderiza o componente Contador corretamente', () => {
  const { getByText } = render(<Contador />);
  const valorElement = getByText('Valor: 0');
  const incrementarButton = getByText('Incrementar');
  const decrementarButton = getByText('Decrementar');

  expect(valorElement).toBeInTheDocument();
  expect(incrementarButton).toBeInTheDocument();
  expect(decrementarButton).toBeInTheDocument();
});

test('incrementa o valor ao clicar no botão de incremento', () => {
  const { getByText } = render(<Contador />);
  const valorElement = getByText('Valor: 0');
  const incrementarButton = getByText('Incrementar');

  fireEvent.click(incrementarButton);

  expect(valorElement).toHaveTextContent('Valor: 1');
});

test('decrementa o valor ao clicar no botão de decremento', () => {
  const { getByText } = render(<Contador />);
  const valorElement = getByText('Valor: 0');
  const decrementarButton = getByText('Decrementar');

  fireEvent.click(decrementarButton);

  expect(valorElement).toHaveTextContent('Valor: 0');
});

test('o valor nunca fica negativo', () => {
  const { getByText } = render(<Contador />);
  const valorElement = getByText('Valor: 0');
  const decrementarButton = getByText('Decrementar');

  fireEvent.click(decrementarButton);

  expect(valorElement).toHaveTextContent('Valor: 0');
});

//parte 3
test('snapshot do componente Contador', () => {
  const tree = renderer.create(<Contador />).toJSON();
  expect(tree).toMatchSnapshot();
});