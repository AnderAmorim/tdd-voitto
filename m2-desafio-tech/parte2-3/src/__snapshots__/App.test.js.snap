// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot do componente Contador 1`] = `
<div>
  <p>
    Valor: 
    0
  </p>
  <button
    onClick={[Function]}
  >
    Incrementar
  </button>
  <button
    onClick={[Function]}
  >
    Decrementar
  </button>
</div>
`;
